### PR TITLE
File association on windows

### PR DESCRIPTION
**Pull request by vQuagliaro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,6 +143,10 @@ Icon to be shown in installation process.
 ### `win.nsiTemplate` *( optional )*
 Option to define a custom NSI installation file.
 
+### `win.fileAssociation` *( optional )*
+Option to define a custom file association on Windows.
+Caution: `win.fileAssociation` does not work when `win.nsiTemplate` is defined.
```
Comment (TristanPouliquen) [inline README.md]: It should still work if the custom template is based on the original one

**Comment by vQuagliaro** [comment]
Hi @stefanjudis, @demetris-manikas and @vxsx,
In the PR, a new option is created to define a custom file association on Windows.
Thank in advance for reviewing and leaving comments.

**Comment by stefanjudis** [comment]
@vQuagliaro sorry for delay.

I'm currently on vacation  but will be back tomorrow. So I'm trying to get to it asap.
If nothing happened in 2day - feel free to ping again. :bowtie:

**Comment by LestaD** [comment]
Very good!

**Comment by vQuagliaro** [comment]
Hi @stefanjudis,
I'm back :smile: !
Thank in advance for reviewing and leaving comments.

**Comment by stefanjudis** [comment]
@vQuagliaro 

Hey hey, I finally had some time to look at it. Sorry again for the delay.

> a custom file association on Windows.

Can you explain me the use case for this. I'm not familiar with this file association thing. :(

**Comment by TristanPouliquen** [comment]
@stefanjudis I worked with Vincent on a project with Electron. 

File association in Windows is the feature that allows you to tell Windows that a `.myapp` file type is to be opened with your Electron application. 

Then, when the user double-clicks on the files with your custom file type, it opens them automatically with your application.

Here is a blog article we wrote on this feature to explain it in more detail: http://www.theodo.fr/blog/2015/12/link-files-to-application-in-windows/

**Comment by stefanjudis** [comment]
@TristanPouliquen @vQuagliaro 

Uhhh nice!!! Thanks for explanation.
I will have a look at it tonight.

What about the macos counterpart. Did you folks play around with that too?

**Comment by TristanPouliquen** [comment]
@stefanjudis It is on our TODO list. But the mechanism seems to be quite different and it is not in the scope of the project we are working on at work for the moment.

But we are building an [`electron-boilerplate`](http://github.com/theodo/electron-boilerplate) to help bootstrap complete Electron applications for all platforms so we will come to it !

**Comment by stefanjudis** [comment]
> But the mechanism seems to be quite different and it is not in the scope of the project we are working on at work for the moment.

Fair enough - probably we would need a proper package installer for that. :)

I will check this PR tonight then and probably also ship a new version of `electron-builder` tonight.

**Comment by stefanjudis** [comment]
Shi....

I was working through the PR's from top. :(
@vQuagliaro Can you please resolve conflicts. 

Sorry.

**Comment by vQuagliaro** [comment]
Yes, I'm resolving conflicts now.

**Comment by vQuagliaro** [comment]
WIP, I'm testing.

**Comment by vQuagliaro** [comment]
@stefanjudis 
Conflicts have been resolved and it has just been tested manually.
The same feature for osx is always on our TODO list.

Thanks for all.

**Comment by vQuagliaro** [comment]
@stefanjudis,
I'm back :smile: !
If you have time, could you please check and merge this PR?

**Comment by stefanjudis** [comment]
LGTM! :+1: 

Just a little side note... please check commit messages of the project. ;)
No need for `[...] foo` here. :) 

But thanks for contribution. :)

**Pull request merged**